packages/xstate-react/test/createActorContext.test.tsx

```diff
@@ -326,4 +326,114 @@ describe('createActorContext', () => {
       expect(screen.getByTestId('value').textContent).toBe('42');
     });
   });
+
+  it('should expect a constant machine reference', () => {
+    const someMachine = createMachine({
+      context: { count: 0 },
+      on: {
+        inc: {
+          actions: assign({ count: ({ context }) => context.count + 1 })
+        }
+      }
+    });
+    const stubFn = jest.fn();
+    const SomeContext = createActorContext(someMachine);
+
+    const Component = () => {
+      const { send } = SomeContext.useActorRef();
+      const count = SomeContext.useSelector((state) => state.context.count);
+      return (
+        <>
+          <span data-testid="count">{count}</span>
+          <button data-testid="button" onClick={() => send({ type: 'inc' })}>
+            Inc
+          </button>
+        </>
+      );
+    };
+
+    const App = () => {
+      return (
+        <SomeContext.Provider
+          logic={someMachine.provide({
+            actions: {
+              testAction: stubFn
+            }
+          })}
+        >
+          <Component />
+        </SomeContext.Provider>
+      );
+    };
+
+    render(<App />);
+
+    expect(screen.getByTestId('count').textContent).toBe('0');
+    fireEvent.click(screen.getByTestId('button'));
+
+    expect(screen.getByTestId('count').textContent).toBe('1');
+
+    fireEvent.click(screen.getByTestId('button'));
+
+    expect(screen.getByTestId('count').textContent).toBe('2');
+  });
+
+  it('options can be passed to the provider', () => {
+    const SomeContext = createActorContext(
+      createMachine({
+        initial: 'a',
+        states: {
+          a: {
+            on: {
+              next: 'b'
+            }
+          },
+          b: {}
+        }
+      })
+    );
+    let state: any = undefined;
+
+    const Component = () => {
+      const { send } = SomeContext.useActorRef();
+      const actorState = SomeContext.useSelector((state) => state);
+
+      state = actorState;
+
+      return (
+        <div
+          data-testid="value"
+          onClick={() => {
+            send({ type: 'next' });
+          }}
+        >
+          {state.value}
+        </div>
+      );
+    };
+
+    const App = () => {
+      return (
+        <SomeContext.Provider options={{ state }}>
+          <Component />
+        </SomeContext.Provider>
+      );
+    };
+
+    const { unmount } = render(<App />);
+
+    expect(screen.getByTestId('value').textContent).toBe('a');
+
+    fireEvent.click(screen.getByTestId('value'));
+
+    expect(screen.getByTestId('value').textContent).toBe('b');
+
+    // unrender app
+    unmount();
+
+    // re-render app
+    render(<App />);
+
+    expect(screen.getByTestId('value').textContent).toBe('b');
```

Comment:
it would be cool to have a comment here to clarify what is even being tested here

Reply:
it feels a little indirect - we can create a persisted state out of the actor, before even rendering a component and then pass that persisted state in the very first render and immediately assert what we want

Reply:
I like doing it as real-life as possible; chances are developers will _not_ construct a state out of the actor manually.

Reply:
I like the "as real-life as possible" approach but I'm not sure that this is it. The user is unlikely to use this kind of remounting in their app. In addition to that... this test gives a false impression to the reader that it's OK to feed the snapshot state like this as persisted state. That's not OK, or at least not OK in some cases. 

You are typing `state` as `any` here (the option is also typed as `any` right now, see [here](https://github.com/statelyai/xstate/blob/69dc9c1a5d14ddc6be9263edc17f4811ddbc1e5e/packages/core/src/types.ts#L1562) so this looks alright-ish. But in fact what you assign to it is of `TSnapshot` type~ whereas what is expected to be provided to the `SomeContext.Provider` is of `TPersistedState` type~. Those are not synonymous and can't be mixed-and-matched - an explicit conversion step should always be required here (and this test should take this into account).

Reply:
Fixed